lib/ProductOpener/Tags.pm

```diff
@@ -782,11 +782,101 @@ sub remove_plurals($$) {
 
 
 
+=head2 sanitize_taxonomy_line( $line )
+
+Sanitize a taxonomy line before processing
+
+=head3 Arguments
+
+=head4 str $line - the line read from the file
+
+=cut
+sub sanitize_taxonomy_line($)
+{
+	my $line = shift;
+ 	
+	chomp($line);
+
+	$line =~ s/’/'/g;  # normalize quotes
+
+	# assume commas between numbers are part of the name
+	# e.g. en:2-Bromo-2-Nitropropane-1,3-Diol, Bronopol
+	# replace by a lower comma ‚
+
+	$line =~ s/(\d),(\d)/$1‚$2/g;
+
+	# replace escaped comma \, by a lower comma ‚
+	$line =~ s/\\,/‚/g;
+
+	# remove parenthesis for roman numerals
+	# fr:E333(iii), Citrate tricalcique
+	# -> E333iii
+
+	$line =~ s/\(((i|v|x)+)\)/$1/i;
+
+	# strip spaces at end of line
+	$line =~ s/\s+$//;
+
+	return $line
+}
+
+
+=head2 search_tag_c( $tag, $synonyms, $tagtype, $lc, $warning )
```

Comment:
I would rename the function to get_lc_tagid(), and change the order of parameters to ($lc, $tagtype, $tag) as it is the order we have in all the other functions like canonicalize_taxonomy_tag()

Reply:
done (next commit)